protocol/x/clob/keeper/deleveraging.go

```diff
@@ -33,13 +32,24 @@ func (k Keeper) MaybeDeleverageSubaccount(
 ) {
 	lib.AssertCheckTxMode(ctx)
 
+	clobPairId, err := k.GetClobPairIdForPerpetual(ctx, perpetualId)
+	if err != nil {
+		return new(big.Int), err
+	}
+	clobPair := k.mustGetClobPair(ctx, clobPairId)
+
 	canPerformDeleveraging, err := k.CanDeleverageSubaccount(ctx, subaccountId)
 	if err != nil {
 		return new(big.Int), err
 	}
 
-	// Early return to skip deleveraging if the subaccount can't be deleveraged.
-	if !canPerformDeleveraging {
+	// Generate final settlement deleveraging matches (at oracle price) if subaccount has non-negative TNC and
+	// the market is in final settlement. This means we will regularly deleverage subaccounts with negative TNC
+	// in markets that are in final settlement.
+	isFinalSettlement := clobPair.Status == types.ClobPair_STATUS_FINAL_SETTLEMENT && !canPerformDeleveraging
+
+	// Early return to skip deleveraging if the subaccount can't be deleveraged. Final settlement deleveraging.
+	if !isFinalSettlement && !canPerformDeleveraging {
```

Comment:
this is slightly confusing to read since you have `!canPerformDeleveraging` at the end of both

Reply:
also nit: I feel like we should move this to `keeper.CanDeleverageSubaccount`. We use the word `Deleverage` to mean both -ve TNC deleveraging and final settlement deleveraging in pretty much all deleveraging functions, so doesn't make sense for `CanDeleverageSubaccount` to be -ve TNC deleveraging only

Reply:
noted, working on updating this